### PR TITLE
Add support for Moonlight 16B, add aux loss for Deepseek v3 model finetuning.

### PR DESCRIPTION
**Pull request by Kimiko-AI**
# What does this PR do?
This add support for moonlight 16B which uses q_lora_rank = null, add sequence-wise auxiliary loss proposed in deepseek v3 paper and this [reimplemtation ](https://github.com/ScienceOne-AI/DeepSeek-671B-SFT-Guide/blob/main/model/DeepSeek-R1-BF16/modeling_deepseek.py)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] I read the contributor guideline, Pull Request section

@ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Very nice! Can you use `modular` to create a new modeling file? As it is different ! 🤗 https://huggingface.co/docs/transformers/en/modular_transformers

Also cab we remove codepathes? 
- `seq_aux` name needs to be better + is this used or not by default for released checkpoint
- `ce` and single letter variables need better naming!
- branches for the loss should be like we do in `Mixtral`: we return the logits / topK indices to be computes outside, no in the module itself! 
- Loss should not be in the model!

**Comment by ArthurZucker** [comment]
cc @Cyrilvallez

**Comment by ArthurZucker** [comment]
Thanks for the new model and PR! 🚀

**Comment by Cyrilvallez** [comment]
Let me know when this is ready for review! 🤗
